src/main/java/org/opensearch/neuralsearch/query/NeuralSparseQueryBuilder.java

```diff
@@ -76,7 +81,12 @@ public class NeuralSparseQueryBuilder extends AbstractQueryBuilder<NeuralSparseQ
     private String queryText;
     private String modelId;
     private Float maxTokenScore;
+    private Map<String, Float> twoPhaseSharedQueryToken;
     private Supplier<Map<String, Float>> queryTokensSupplier;
+    // A parameter that can detect if twoPhase are enabled, when twoPhasePruneRatio equals -1f, it means it two-phase rescoreQueryBuilder's
+    // subQueryBuilder.
+    private float twoPhasePruneRatio = 0F;
```

Comment:
It's twoPhaseSharedQueryToken that used to determine if twoPhase are enabled, right? The comments here seems not very precise

Reply:
There's 3 case , main querybuilldr in 2-phase, sub querybuilder in 2-phase, main querybuilder without 2-phase, 
which 2 case will have a null twoPhaseSharedQueryToken. So twoPhasePruneRatio will help to determine.

Reply:
But if we make low score tokens always not null, we can not consider using this filed to determine any more.

Reply:
> which 2 case will have a null twoPhaseSharedQueryToken. So twoPhasePruneRatio will help to determine.

Do you mean sub querybuilder in 2-phase, main querybuilder without 2-phase? Do we have any logics handle them differently here?

Reply:
Yes, twoPhasePruneRatio
1.0f-> 2-phase disabled
(0f,1f) -> 2-phase enabled and this is the main neural sparse querybuilder
(-1f,-0f) -> 2-phase enabled and this is the sub neural sparse querybuilder.

Reply:
> 2-phase enabled and this is the sub neural sparse querybuilder
For this case it's always -1 right? Why can it has value from (-1f,-0f)

I understand it can distinguish this case, but why we need to distinguish them here? From the two phase processor we can distinguish them without this twoPhasePruneRatio. From the NeuralSparseQueryBuilder I don't  find the logics we need to handle them differently

Reply:
Yes, we don't need to distinguish now, but we need this field to split query tokens in rewrite, and I think make it carry more information maybe better? 
For example, if in a future version, we can find neural sparse query which not be accelerated, whose ratio is 1.0, can make log very easy. And for further need, I think carry more information is better?
Do you think it may make confuse for developer?

Reply:
I think we can add more comments about this. e.g. the meaning of each cases, and we don't need these information now, but keep them for future extension etc

Reply:
Current comments only mentions twoPhasePruneRatio, but in line 333 it checks twoPhaseSharedQueryToken for whether 2-phase is enbaled. It's confusing for me

Reply:
> 1.0f-> 2-phase disabled

But the default value is 0F?

Reply:
> > 1.0f-> 2-phase disabled
> 
> But the default value is 0F?

0.0f-> 2-phase disabled
(0f,1f) -> 2-phase enabled and this is the main neural sparse querybuilder
-1f -> 2-phase enabled and this is the sub neural sparse querybuilder.
Sorry, this is the logic,
And I make some change to let it carry more information.
0.0f-> 2-phase disabled
ratio value (0f,1f) -> 2-phase enabled and this is the main neural sparse querybuilder
-1*ratio value (-1f,0f) -> 2-phase enabled and this is the sub neural sparse querybuilder.

And for your confusing, these information are not required now, the ratio just work for split tokens.